File: ts/twoSum.ts
function twoSum(numbers: number[], target: number): number[] {
  let i = 0;
  let j = numbers.length - 1;

  while (i < j) {
    const sum = numbers[i] + numbers[j];

    if (sum === target) {
      return [i + 1, j + 1];
    } else if (sum < target) {
      i += 1;
    } else {
      j -= 1;
    }
  }

  return [1, 2];
}

export default () => {
  // const numbers = [-1, 0];
  // const target = -1;
  const numbers = [2, 3, 4];
  const target = 6;
  // const numbers = [2, 7, 11, 15];
  // const target = 9;
  const result = twoSum(numbers, target);

  console.log(result);
};

File: rs/src/two_sum.rs
fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
    let mut i = 0;
    let mut j = numbers.len() - 1;
    let mut result = vec![1, 2];

    while i < j {
        let sum = numbers[i] + numbers[j];

        if sum == target {
            result[0] = i as i32 + 1;
            result[1] = j as i32 + 1;
            break;
        } else if sum < target {
            i += 1;
        } else {
            j -= 1;
        }
    }

    return result;
}

pub fn main() {
    let numbers = vec![2, 3, 4];
    let target = 6;

    let result = two_sum(numbers, target);
    println!("{:?}", result);
}
